plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java

```diff
@@ -557,9 +570,21 @@ private Instance createInstance(final String containerName,
                                                                 .withHostConfig(hostConfig)
                                                                 .withEnv(env.toArray(new String[env.size()]));
 
-            final String containerId = docker.createContainer(config, containerName).getId();
+            final String containerId = docker.createContainer(CreateContainerParams.create(config)
+                                                                                   .withContainerName(containerName))
+                                             .getId();
+
+            docker.startContainer(StartContainerParams.create(containerId));
 
-            docker.startContainer(containerId, null);
+            machineLogsStreamer.execute(() -> {
+                try {
+                    docker.attachContainer(AttachContainerParams.create(containerId)
```

Comment:
When TCP connection implementation is used connection can be dropped due to timeout . Make sure we process this situation correctly.